ONSdigital/eq-author-app
# 1183 validating on earliest and latest date range

### What is the context of this PR?

Previously, When adding a date range answer - there was no validation occurring on date ranges and earliest /latest dates.
this PR is to fix validation on these

### How to review 

1. Create a new questionnaire and add a new question with 'Date Range' answer
2. Add an Earliest Date that is AFTER a Latest Date - you should get a validation message.
3. Add a Latest Date that is BEFORE an Earliest Date - you should get a validation message.
4. Correct these dates and the validation message should be gone!

5. Play around with 'Earliest Date is' XX days/months or years BEFORE  on 'Earliest Date.
this should trigger a validation message if that then makes the earliest date AFTER the latest date.

6. Do the same (ViseVersa) for 'Latest Date After' field.

EG - Earliest Date is 2nd Jan 2020 and latest date is 1st Jan 2020 - INVALID MSG
add 2 days or more to the 'Earliest date is before' field and validation message should clear.

EG - Earliest Date is 2nd Jan 2020 and latest date is 1st Jan 2020 - INVALID MSG
add 2 days or more to the 'Latest Date is After' field and validation message should clear.

There is currently NO validation on Duration Fields as that is part of another ticket!!
also - duration fields should NOT affect the validation on date range (currently) at all.

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process
